arena.go

```diff
@@ -47,55 +45,90 @@ type Arena interface {
 	Release()
 }
 
+// singleSegmentPool is a pool of *SingleSegmentArena.
+var singleSegmentPool = sync.Pool{
```

Comment:
I think we still want some kind of bucketing by size, similar to what we have in [exp/bufferpool](https://github.com/capnproto/go-capnp/blob/main/exp/bufferpool/pool.go#L35).  A pernicious form of resource leak occurs when a large buffer is allocated and used once, and then kept alive by `sync.Pool` only to back very _small_ segments.

Reply:
This only buffers `SingleSegmentArena` references (_not_ their internal data). The internal data slice is still cached by the global pool.

Reply:
Ah right! 💡 
SGTM, resolving.